FWCore/Framework/interface/CallbackExternalWork.h

```diff
@@ -193,7 +200,29 @@ namespace edm {
                                           }));
       }
 
+      template <typename U>
+      struct AcquireCacheType {
```

Comment:
It would be better to move these class definitions outside of the `CallbackExternalWork` class (e.g. in `edm::eventsetup::impl` namespace) so that the compiler would not have to generate code for them for every instantiation of `CallbackExternalWorker`.

Reply:
Done. Thanks.